File: JanoMotorsApi/JanoMotorsApi/Models/UserInfo.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace JanoMotorsApi.Models
{
    public class UserInfo
    {
        public int Id { get; set; }
        [StringLength(13)]
        public required string PhoneNumber { get; set; }
        [MinLength(1)]
        public required string Firstname { get; set; }
        [MinLength(1)]
        public required string Lastname { get; set; }
        public required DateOnly DateOfBirth { get; set; }
        public byte[]? ProfileImage { get; set; }

        [JsonIgnore]
        public User User { get; set; } = null!;
    }
}

File: jano_motors_app/src/app/Interfaces/user-info.ts
import { UserDocPost } from "./user-doc";
export interface UserInfoPost {
    id: number;
    firstName: string;
    lastName: string;
    dateOfBirth: string;
    phoneNumber: string;
    profileImage: string;
    idFrontImage: File;
    idBackImage: File;
    driverLicenseFrontImage: File;
    driverLicenseBackImage: File;
}

export interface UserInfo {
    id: number;
    firstname: string;
    lastname: string;
    dateOfBirth: Date;
    phoneNumber: string;
    profileImage: string;
}
